src/store/utility.ts

```diff
@@ -173,6 +174,96 @@ function skipMemoText(targettext: string): string {
   return resolvedText;
 }
 
+/**
+ * 新しく変更されるアクセント句に対して、変更前のモーラを適用するクラス
+ *
+ * まず、過去と現在のアクセント句から、モーラ配列を作成する。
+ * モーラ配列の各テキストを配列にまとめ、得られた2つの配列より、差分を検出する。
+ * 検出した差分の追加、削除、それ以外の状況に応じて、変更前のモーラ配列に対して、文字列の挿入、配列自体の削除
+ * を適用し、変更前のモーラ配列のテキストと変更後のアクセント句のモーラ配列のテキストを一致するようにする。
+ * 最後に、変更後のアクセント句を走査して、変更前のモーラ配列のオブジェクト部分を変更後のアクセント句のモーラに適用する。
+ */
+export class AccentDiff {
+  beforeAccent: AccentPhrase[];
+  afterAccent: AccentPhrase[];
+  constructor(beforeAccent: AccentPhrase[], afterAccent: AccentPhrase[]) {
+    this.afterAccent = JSON.parse(JSON.stringify(afterAccent));
+    this.beforeAccent = JSON.parse(JSON.stringify(beforeAccent));
+  }
+
+  createFlatArray(collection: AccentPhrase[], Key: string) {
+    const targetAccent = JSON.parse(JSON.stringify(collection));
+    const result = [];
+    for (const element of targetAccent) {
+      result.push(element[Key]);
+    }
+    return result.flat();
+  }
+
+  /**
+   * モーラのパッチ配列を作成するメンバ関数
+   */
+  createDiffPatch() {
+    const after = JSON.parse(JSON.stringify(this.afterAccent));
+    const before = JSON.parse(JSON.stringify(this.beforeAccent));
```

Comment:
```suggestion
    const after = structuredClone(this.afterAccent);
    const before = structuredClone(this.beforeAccent);
```
[structuredClone](https://developer.mozilla.org/ja/docs/Web/API/structuredClone) でも良いと思います。

Reply:
これ初めて知りました。ありがとうございます。